fv3core/decorators.py

```diff
@@ -330,11 +332,37 @@ def decorator(func):
     return decorator
 
 
-class FixedOriginStencil:
-    """Wrapped GT4Py stencil object explicitly genrating
+class StencilWrapper:
+    """Wrapped GT4Py stencil object."""
+
+    def __init__(self, func: Callable, **kwargs):
+        self.func = func
+
+        if "format_source" not in kwargs:
+            kwargs["format_source"] = global_config.get_format_source()
+        gt4py_utils.apply_device_sync(kwargs)
+
+        self.stencil_object = gtscript.stencil(
+            backend=global_config.get_backend(),
+            rebuild=global_config.get_rebuild(),
+            definition=self.func,
+            **kwargs,
+        )
+
+    def __call__(self, *args, **kwargs) -> None:
+        self.stencil_object(
+            *args,
+            **kwargs,
+            validate_args=global_config.get_validate_args(),
+        )
+
+
+class FrozenStencil(StencilWrapper):
```

Comment:
I like this name, since we have fixed domains too now

Reply:
Thank you, it was a combined effort with @twicki .